contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/print_statements.py

```diff
@@ -16,6 +18,10 @@ class PrintStatements(CheckstylePlugin):
   FUNCTIONY_EXPRESSION = re.compile(r'^\s*\(.*\)\s*$')
 
   def nits(self):
+    if PY3:
+      # Python 3 interpreter will raise SyntaxError upon reading a print statement.
+      # So, this module cannot be meaningfully used when ran with a Python 3 interpreter.
+      return
```

Comment:
I'm not sure what to do about this. There is no way for Python 3 to parse a print statement, even if it's reading Python 2.

To reproduce, open Python 3 console:
```python
import ast
ast.dump(ast.parse('print "hi"'))
```

I'm concerned this implies once we switch Pants to Python 3 under-the-hood, it will start to complain about Python 2 only syntax like print statements and `except Error, e` syntax? If so, this means users still on Python 2 will have to convert all Py2-only syntax to Py2-3 compliant to use the newest version of Pants. I see no way around this, and think it is fairly acceptable because they do not have to fully migrate to Python 3. 

Still, we should communicate well with users about what must be changed and how to use the futurize script to do it easily.

Reply:
cc @CMLivingston has had some thoughts about this, which @jsirois recently incorporated: see how #6182 dynamically creates a pex to run a linter.

Reply:
Interesting. So before we release a stable version of Pants running Python 3, we could create a Pex to run this contrib code using Py2 interpreter?

There's a counterpoint, the Python 2 interpreter will complain about Python 3 only syntax, like type hints
```python
# Python2
ast.dump(ast.parse('x: int = 4'))
```

So, I don't think there's a way we could support linting for both Python 2-only syntax and Python 3-only syntax? We could get fancy and use the `compatibility` flag to trigger the corresponding linter for that module, perhaps. The common subset of Python 2 and Python 3 of course will work as well, like running Python 3 linter over Pants in its current Python 2 state.

It seems to me that unless we had two PEXes and dynamically chose which to use depending on the file, we'll have to choose between supporting Py 2 only + Py 2/3 subset, or Py3 only + Py 2/3 subset.

Reply:
Dynamically creating the linter pex means you don't need to "N" pexes for N python versions... just linter code that is compatible with all of the (major) python versions you want to lint. You'd then dynamically create a pex for python 3 if you needed to lint 3, etc.